new/server-enforced-recipe.md

```diff
@@ -0,0 +1,104 @@
+---
+RFC: unassigned
+Title: Server Enforced Recipe
+Author: Daniel DeLeo <dan@chef.io>
+Status: Draft
+Type: Standards Track
+<Tracking:>
+<  - https://github.com/chef/chef/issues/X>
+---
+
+# Server Enforced Recipe
+
+## Description
+
+Chef Server will provide an endpoint that MAY serve a Chef recipe file. Chef
+Client will attempt to fetch the recipe during run context setup. If
+no user action is taken to configure the feature, the endpoint returns 404
+and Client behavior will be unaffected. When the feature is enabled, the
+endpoint returns the configured recipe file. Chef Client will evaluate and
+converge the recipe.
+
+## Rationale
+
+The motivation for this feature is to allow the operator of the Chef Server to
+enforce desired client-side configuration using Chef. Intended use cases include:
+
+* Allow cloud-based vendors to install an agent necessary for correct operation
+  of the service
+* Allow Chef Customer Development Partners to efficiently install experimental
+  client-side software during feature development
+* Allow organizations that operate as internal service providers to enforce
+  standard configurations
+
+Currently the only way to achieve these goals with Chef is to enforce the
+inclusion of recipes on a node's run list, which is not feasible in many cases.
+
+The enforced policy is limited to a single recipe instead of a full cookbook or
+secondary run list for several reasons:
+
+* Cookbooks are Chef Server objects that are organization-scoped and subject to
+  authorization restrictions. Allowing some cookbooks to be global requires
+  additional complexity which is not needed for the intended uses.
+* Cookbooks have versions and dependencies, which have to be solved. There are
+  several ways this could be addressed, but all options introduce unneeded
+  complexity into the solution.
+
+## Motivation
+
+    As a Chef Server Service Provider,
+    I want to enforce a recipe to run on client systems,
+    so that I can ensure client systems are correctly configured.
+
+## Specification
+
+### Enforced Recipe Endpoint
+
+Chef Server shall expose an organization-scoped endpoint for the enforced
+recipe. If the feature has not been configured by the Chef Server
+administrator, the endpoint shall return a 404 response. If the feature is
+enabled by the Chef Server administrator, the endpoint shall return a 200
```

Comment:
It would probably be good for future expansion to include some node details, maybe the node name? I could see an argument for platform+platform_version but conditionals like that should probably just be handled in the recipe code itself.